### PR TITLE
Remove all listeners after emitting error in RequestHeaderParser

### PR DESCRIPTION
**Pull request by WyriHaximus**
Follow up on @clue's comment here: https://github.com/reactphp/http/pull/65#discussion_r78152932

### CODE REVIEW

**Comment by WyriHaximus** [comment]
Ping @clue

File: src/RequestHeaderParser.php [hunk 1]
```diff
@@ -29,7 +29,7 @@ public function feed($data)
             try {
                 $this->parseAndEmitRequest();
             } catch (Exception $exception) {
-                $this->emit('error', [$exception]);
+                $this->emit('error', [$exception, $this]);
```
Comment (clue) [inline src/RequestHeaderParser.php]: Not a fan of this tbh. This should probably be discussed in the Stream component instead, but this messes with forwarding events.

Comment (WyriHaximus) [inline src/RequestHeaderParser.php]: What would you suggest? Remove it here and from line 21, which would create a minor BC break. Or something else?

Comment (WyriHaximus) [inline src/RequestHeaderParser.php]: Tbh I have no problem reverting this line and associated tests. This line hasn't caused any issues. I'll revert it :+1:

**Comment by WyriHaximus** [comment]
Not sure why this PR isn't seeing my latest changes. Contacting github

**Comment by WyriHaximus** [comment]
Turned out I committed it to my fork instead of the original repo here :blush:

**Pull request merged**